compiler_opt/rl/corpus.py

```diff
@@ -14,12 +14,105 @@
 # limitations under the License.
 """ModuleSpec definition and utility command line parsing functions."""
 
+from absl import logging
 from dataclasses import dataclass
+from typing import Iterable, Optional, Tuple, List
+
+import os
+import tensorflow as tf
 
 
 @dataclass(frozen=True)
 class ModuleSpec:
   """Dataclass describing an input module and its compilation command options.
   """
   name: str
-  has_thinlto: bool = False
+  _exec_cmd: Tuple[str, ...] = ()
+
+  def cmd(self) -> List[str]:
```

Comment:
why not just leave the exec_cmd publicly accessible - the object is frozen anyway?

Reply:
To make sure that everything outside uses `.cmd()`,  so later on more stuff can be added on via `.cmd()` without having to change every reference of `.exec_cmd` -> `.cmd()`

Reply:
but do we want to add more stuff to cmd? I'm assuming that's about stuff like substitutions. I'd argue those belong outside ModuleSpec - as a stand-alone command line converter or something like that. Basically, to compile, you need an array that's appropriately contextualized to the local location the input(s) and profile file(s) are, model under training, extra domain-specific flags... so a (ModuleSpec, extra options) -> array converter. For separation of concerns reasons, the ModuleSpec can stay agnostic of all that, as a logical holder of the related information that can generate that command line. 

My suggestion is to leave ModuleSpec "dumb", basically - so then you can have exec_cmd as a ctor - initialized frozen property.